pkgs/dart_mcp/lib/src/api/completion.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'api.dart';
+
+// /* Autocomplete */
+// /**
+//  * A request from the client to the server, to ask for completion options.
+//  */
+// export interface CompleteRequest extends Request {
+//   method: "completion/complete";
+//   params: {
+//     ref: PromptReference | ResourceReference;
+//     /**
+//      * The argument's information
+//      */
+//     argument: {
+//       /**
+//        * The name of the argument
+//        */
+//       name: string;
+//       /**
+//        * The value of the argument to use for completion matching.
+//        */
+//       value: string;
+//     };
+//   };
+// }
+
+// /**
+//  * The server's response to a completion/complete request
+//  */
+// export interface CompleteResult extends Result {
+//   completion: {
+//     /**
+//      * An array of completion values. Must not exceed 100 items.
+//      */
+//     values: string[];
+//     /**
+//      * The total number of completion options available. This can exceed the
+//      * number of values actually sent in the response.
+//      */
+//     total?: number;
+//     /**
+//      * Indicates whether there are additional completion options beyond those
+//      * provided in the current response, even if the exact total is unknown.
+//      */
+//     hasMore?: boolean;
+//   };
+// }
+
+// /**
+//  * A reference to a resource or resource template definition.
+//  */
+// export interface ResourceReference {
+//   type: "ref/resource";
+//   /**
+//    * The URI or URI template of the resource.
+//    *
+//    * @format uri-template
+//    */
+//   uri: string;
+// }
+
+/// Identifies a prompt.
+extension type PromptReference.fromMap(Map<String, Object?> _value) {
```

Comment:
should this be in a part file prompts.dart?

Reply:
It is only used as a part of completions actually - so I think it makes sense to be here. I accidentally ported it early as a part of Prompts support not realizing its really a part of completions support.